File: src/main/scala/com/abaddon83/cardsGames/burracoGames/ports/PlayerPort.scala
package com.abaddon83.cardsGames.burracoGames.ports

import com.abaddon83.cardsGames.burracoGames.domainModels.burracoGames.initialised.playerInGames.PlayerInGame
import com.abaddon83.cardsGames.burracoGames.domainModels.{BurracoPlayer, PlayerNotAssigned}
import com.abaddon83.cardsGames.shares.players.PlayerIdentity

import scala.concurrent.Future

trait PlayerPort {
  def findPlayerNotAssignedBy(playerIdentity: PlayerIdentity): Future[PlayerNotAssigned]
  def findBurracoPlayerBy(playerIdentity: PlayerIdentity): Future[BurracoPlayer]
  def findBurracoPlayerInGameBy(playerIdentity: PlayerIdentity): Future[PlayerInGame]
}

File: src/main/kotlin/com/abaddon83/burracoGame/ports/PlayerPort.kt
package com.abaddon83.burracoGame.ports

import com.abaddon83.burracoGame.domainModels.BurracoPlayer
import com.abaddon83.burracoGame.domainModels.PlayerNotAssigned
import com.abaddon83.burracoGame.domainModels.burracoGameExecutions.playerInGames.PlayerInGame
import com.abaddon83.burracoGame.shared.players.PlayerIdentity

interface PlayerPort {
        suspend fun findPlayerNotAssignedBy(playerIdentity: PlayerIdentity): PlayerNotAssigned
        suspend fun findBurracoPlayerBy(playerIdentity: PlayerIdentity): BurracoPlayer
        suspend fun findBurracoPlayerInGameBy(playerIdentity: PlayerIdentity): PlayerInGame
}